src/cmd/tools/verify_data_files/main/main.go

```diff
@@ -396,39 +394,9 @@ func fixFileSet(
 			return err
 		}
 
-		// Need to save tags in case we need to write them out again
-		// (iterating them in read entry means we can't reiterate them
-		// without copying/duplicating).
-		currTags = currTags[:0]
-		for tags.Next() {
-			tag := tags.Current()
-			name := tag.Name.Bytes()
-			value := tag.Value.Bytes()
-
-			// Need to take copy as only valid during iteration.
-			nameCopy := opts.bytesPool.Get(len(name))
-			nameCopy.IncRef()
-			nameCopy.AppendAll(name)
-			valueCopy := opts.bytesPool.Get(len(value))
-			valueCopy.IncRef()
-			valueCopy.AppendAll(value)
-			copies = append(copies, nameCopy)
-			copies = append(copies, valueCopy)
-
-			currTags = append(currTags, ident.Tag{
-				Name:  ident.BytesID(nameCopy.Bytes()),
-				Value: ident.BytesID(valueCopy.Bytes()),
-			})
-		}
-
-		// Choose to write out the current tags if do not need modifying.
-		writeTags := currTags[:]
+		tagsCopy := tags.Duplicate()
 
-		var currIdentTags ident.Tags
-		currIdentTags.Reset(currTags)
-		currTagsIter.Reset(currIdentTags)
-
-		check, err := readEntry(id, currTagsIter, data, checksum)
+		check, err := readEntry(id, tags, data, checksum)
```

Comment:
nit: I think it's a bit clearer if this is the `tagsCopy` and the iterator is built on top of the base tags?

Reply:
Sure yeah.